File: BOJ/Greedy/2839/2839.py
# 2839 설탕 배달
# https://www.acmicpc.net/problem/2839
import sys
input = sys.stdin.readline

n = int(input()) #설탕 무게입력
ans = 0
while True:
    if n%5 == 0:
        ans+=(n//5)
        print(ans)
        break
    n -= 3
    ans += 1
    if n<0:
        print(-1)
        break

File: BOJ/Greedy/2839/2839.cpp
// 2839 설탕 배달
// https://www.acmicpc.net/problem/2839
#include <iostream>

using namespace std;

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int N, answer = 0;

    cin >> N;

    while (N > 0) {
        if (N % 5 == 0) {
            answer += (N / 5);
            break;;
        } else {
            N -= 3;
            answer++;
            if (N < 0) {
                answer = -1;
            }
        }
    }
    
    cout << answer;

    return 0;
}